server/service/apple_mdm.go

```diff
@@ -1979,12 +1979,97 @@ func (e *APNSDeliveryError) Unwrap() error { return e.Err }
 
 func (e *APNSDeliveryError) StatusCode() int { return http.StatusBadGateway }
 
+// ensureFleetdConfig ensures there's a fleetd configuration profile in
+// mdm_apple_configuration_profiles for the AppConfig.MDM.AppleBMDefaultTeam or
+// for hosts in no team.
+//
+// We try our best to use AppConfig.MDM.AppleBMDefaultTeam but we default to creating a profile for "no team" if:
+//
+//   - AppConfig.MDM.AppleBMDefaultTeam is empty.
+//   - A team with the name provided in the config can't be found.
+//   - A team is found, but it doesn't have any enroll secrets.
+//
+// Also, the following is assumed:
+//
+//   - This profile will be applied to all hosts in the team (or "no team",)
+//     but it will only be used by hosts that have a fleetd installation without
+//     an enroll secret and fleet URL (mainly DEP enrolled hosts).
+//   - Once the profile is applied to a team (or "no team",) it's not removed if
+//     AppConfig.MDM.AppleBMDefaultTeam changes, this is to preserve existing
+//     agents using the configuration (mainly ServerURL as EnrollSecret is used
+//     only during enrollment)
```

Comment:
@gillespi314 @mna I want to bring your attention to this comment, please let me know if it makes sense or if you have other ideas.

Reply:
The comment makes sense to me, it's just a bit unclear why this section applies:

> // This profile will be applied to all hosts in the team (or "no team",)
> //     but it will only be used by hosts that have a fleetd installation without
> //     an enroll secret and fleet URL (mainly DEP enrolled hosts).

I think that's because hosts that enroll with an enroll secret will have their enrollment profile automatically generated during orbit enrollment? Or is it because they are expected to manually enroll? I'd clarify that in the comment (why is it just for fleet without enroll secret, what happens for the other hosts).